packages/cli/src/services/template-renderer.ts

```
@@ -97,3 +92,178 @@ export function expandLoopVariables(
+ * @param variables - Key-value pairs for substitution
+ * @param escape - Optional escape function applied to values before substitution
+ * @returns Text with matched variables replaced
+ */
+export function substituteText(
+  text: string,
+  variables: Record<string, string>,
+  escape?: (value: string) => string
+): string {
+  return text.replace(TEMPLATE_VAR_REGEX, (match, name: string) => {
+    if (!Object.prototype.hasOwnProperty.call(variables, name)) return match;
+    const value = variables[name];
```

Biome (2.3.14)

[error] Do not shadow the global "escape" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)